backend/state.py

```diff
@@ -232,6 +239,49 @@ def update_special_effect(self, special_effect, arg, param_arg_dict):
         current = self.special_effects[self.special_effects.index(replaced_effect)]
         current.update(self, active=True)
 
+    def _get_next_ID_for_object(self, obj):
+        """
+        Gets the next available ID for an object.
+
+        Args:
+            obj (Object): The object to get the next available ID for.
+
+        Returns:
+            int: The next available ID for the object.
+        """
+        num = 0
+        for object in self.objects:
+            name, id = trim_item_ID(object.name)
+            if name == obj.name:
+                num = max(num, id)
+        return num+1
+
+    def add_object(self, obj):
+        """
+        Adds an object to the state.
+
+        Args:
+            obj (Object): The object to add to the state.
+        """
+        num = self._get_next_ID_for_object(obj)
+        obj.name = f"{obj.name}{num}"
```

Comment:
This works because in the creation effect, add_object() is called on the field self.created_obj, which has a name with no id (such as lettuce, and not lettuce1) because we don't yet know what the id of the obj is when defining it in the json. 

TODO added